### PR TITLE
[ENHANCE 🗣] (Sub)category name on cards

### PR DESCRIPTION
**Pull request by eoji**
# what
- Displaying subcategory when viewing root categories or all projects(including staff picks, saved, recommendations, and backed by people you follow).
- Displaying category when a project's category is a root category. (but hiding if you're looking at a root category filter)
- Displaying `Projects We Love` when project is a staff pick but hiding when viewing the staff picks filter.
- Hiding both potential tags when project is not a staff pick and you're filtered by the project's category.
- Added documentation to the outputs of `ProjectCardHolderViewModel` since there wasn't any

# see
All Projects &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;  Staff Picks/Projects We Love

 
Root category &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Subcategory
 
Added documentation for 

# me
image

### CODE REVIEW

**Comment by eoji** [comment]
[Show category on discovery project card](https://trello.com/c/zsKDiE4n/1013-show-category-on-discovery-project-card)

File: app/src/main/java/com/kickstarter/viewmodels/DiscoveryFragmentViewModel.java [hunk 1]
```diff
@@ -212,6 +216,14 @@ public ViewModel(final @NonNull Environment environment) {
         .subscribe(p -> this.koala.trackViewedUpdate(p, KoalaContext.Update.ACTIVITY_SAMPLE));
     }
 
+    private List<Pair<Project, DiscoveryParams>> combineProjectsAndParams(final @NonNull List<Project> projects, final @NonNull DiscoveryParams params) {
+      final ArrayList<Pair<Project, DiscoveryParams>> projectAndParams = new ArrayList<>(projects.size());
+      for (int i = 0; i < projects.size(); i++) {
```
Comment (Rcureton) [inline app/src/main/java/com/kickstarter/viewmodels/DiscoveryFragmentViewModel.java]: Haven't used one of these in a while 🙃

File: app/src/main/java/com/kickstarter/viewmodels/ProjectCardHolderViewModel.java [hunk 2]
```diff
@@ -26,49 +27,125 @@
...
   interface Inputs {
     /** Call to configure view model with a project. */
-    void configureWith(Project project);
+    void configureWith(Pair<Project, DiscoveryParams> projectAndDiscoveryParams);
 
     /** Call when the project card has been clicked. */
     void projectCardClicked();
   }
 
   interface Outputs {
```
Comment (Rcureton) [inline app/src/main/java/com/kickstarter/viewmodels/ProjectCardHolderViewModel.java]: Documentation!!!!!!!

File: app/src/main/java/com/kickstarter/viewmodels/ProjectCardHolderViewModel.java [hunk 3]
```diff
@@ -194,9 +284,20 @@ public ViewModel(final @NonNull Environment environment) {
         .map(Project::stateChangedAt)
         .compose(coalesce(new DateTime()));
 
-      this.rootCategoryNameForFeatured = this.project
-        .map(Project::category)
-        .filter(ObjectUtils::isNotNull)
+      this.projectWeLoveIsGone = this.project
+        .map(Project::staffPick)
+        .compose(coalesce(false))
+        .compose(combineLatestPair(this.discoveryParams.map(DiscoveryParams::staffPicks).compose(coalesce(false))))
```
Comment (Rcureton) [inline app/src/main/java/com/kickstarter/viewmodels/ProjectCardHolderViewModel.java]: Why is the `coalese(false)` opposed to taking in a value?

Comment (eoji) [reply app/src/main/java/com/kickstarter/viewmodels/ProjectCardHolderViewModel.java]: `staffPick` is `@Nullable`. If it is null, we want to force it to be false.

**Pull request merged**